### PR TITLE
[Hackday][FrameworkBundle] RedirectController::redirect preserve query params

### PR DESCRIPTION
**Pull request by mxnr**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #9242 |
| License | MIT |
| Doc PR | n/a |

Changed RedirectController::redirect to preserve query parameters. This behaviour is consistent with RedirectController::urlRedirect

### CODE REVIEW

**Comment by mxnr** [comment]
@jakzal i think it can be useful to add some param like `ignoreAttributes `, maybe ~`ignoreQueryParams`? It can also handle bool or array and merge or exclude some query params.
What do you think?

**Comment by backbone87** [comment]
i dont think you can reuse the router candy, that is adding unknown route params as QS params in a generic way, because it would cause problems if your QS contains a param with the same name as a route param

**Comment by fabpot** [comment]
ping @symfony/deciders, I'm not convinced that this should be done at all.

**Comment by weaverryan** [comment]
Sure, I don't like the inconsistency, but I was actually surprised that *either* of these methods keeps the query parameters, as that's not the behavior I'd expect from saying "redirect to /foo". I'd rather not add the query string ever, but that's not a feasible BC change. The only way to make it consistent is this change, which also changes the behavior (but in the safer direction).

But is it really a problem? The original issue just cites the behavior difference, but nobody is specifically saying they need this (and it's super easy to implement yourself). We could just doc it, and tell them if they need to keep the query string or anything else, just add it yourself.

File: src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ public function redirectAction(Request $request, $route, $permanent = false, $ig
 
         $attributes = array();
         if (false === $ignoreAttributes || is_array($ignoreAttributes)) {
-            $attributes = $request->attributes->get('_route_params');
+            $attributes = array_merge($request->attributes->get('_route_params'), $request->query->all());
```
Comment (weaverryan) [inline src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php]: If this were accepted, I think the implementation should match what's done in `urlRedirectAction` - i.e. don't merge the query params into the attributes, but instead append the `?` or `&` at the end with the `$request->getQueryString()` method return.

**Comment by fabpot** [comment]
Closing for the reasons given by @weaverryan

**Pull request closed without merging**